### PR TITLE
tools: added nix flake for convenience

### PR DESCRIPTION
**Pull request by TheButlah**
I use flakes to manage developer shells, so I figured I'd add a simple flake for setting up pyocd, protobuf, etc.

### CODE REVIEW

File: flake.nix [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+          # The other platforms we support for cross compilation
+          arm-linux = nixpkgs.legacyPackages.aarch64-linux;
+          x86-linux = nixpkgs.legacyPackages.x86_64-linux;
+          arm-macos = nixpkgs.legacyPackages.aarch64-darwin;
+          x86-macos = nixpkgs.legacyPackages.x86_64-darwin;
+        };
+        pythonShell = (ps: with ps; [
+          pyocd
+          cmsis-pack-manager
+          cffi
```
Comment (fouge) [inline flake.nix:28]: I don't think we need these

**Review by fouge — CHANGES_REQUESTED** [review]
A conda environment file is available in `orb/utils/env/environment.yml`, can we move the flake files to this location as well? I know it's probably not the standard way to do it, but I don't want to bloat the root directory with files that aren't going to be used by most of the developers. 

I would accept putting this file in the root directory only if it brings everything that's needed to build the firmware, and I would love to use nix in this repo, so if you have some time, we can try to bring the full environment with nix.

File: flake.nix (hunk 1 above)
Comment (TheButlah) [reply flake.nix:28]: they are necessary for pyocd

**Comment by TheButlah** [comment]
Moved filed into `utils/env` folder. You can `cp utils/env/.envrc.example .envrc` to source the flake.

**Review by fouge — APPROVED** [review]

**Pull request merged**